### PR TITLE
[CP-1686] handle downloading multiple releases

### PR DESCRIPTION
**Pull request by wojcik92michal**
Jira: [CP-1686]

**Description**

Scope:
* added logic for downloading multiple releases

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1686]: https://appnroll.atlassian.net/browse/CP-1686?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/__deprecated__/renderer/styles/theming/theme.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ const theme = {
       tabHover: grey1,
       actionHover: blue1,
       warning: orange,
+      info: grey1,
```
Comment (dkarski) [inline packages/app/src/__deprecated__/renderer/styles/theming/theme.ts:47]: We are sure that we would like to add a new color to text? 😬 I recommended getting a similar one from the existing one and giving feedback to a design team. 

Link to the Fonts section in our design system 👉 https://www.figma.com/file/82xELWhdkYtGeDQRkxYnf3/Center---Design-System?node-id=1%3A3&t=Nl9HvRhsCZQxZ17f-0

File: packages/app/src/overview/components/system/system.component.tsx [hunk 2]
```diff
@@ -119,21 +122,27 @@ const System: FunctionComponent<Props> = ({
...
               />
             )
           ) : (
             <CardActionButton
               active
               labelMessage={messages.systemCheckForUpdates}
-              onClick={onUpdateCheck}
+              // AUTO DISABLED - fix me if you like :)
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+              onClick={() => onUpdateCheck()}
```
Comment (dkarski) [inline packages/app/src/overview/components/system/system.component.tsx:145]: Why this change with an arrow function is needed here? 🤔

**Review by dkarski — APPROVED** [review]
The PR it's ok ✅ 

Just have a few questions. Please don't force yourself to answer me before merging if you would like to.

File: packages/app/src/update/actions/download-updates/download-updates.action.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  {
+    rejectValue: AppError<UpdateError>
+  }
+>(
+  UpdateOsEvent.DownloadUpdate,
+  async ({ releases }, { getState, rejectWithValue, dispatch }) => {
+    const { device } = getState() as RootState & ReduxRootState
+    const batteryLevel = device.data?.batteryLevel ?? 0
+
+    if (!isBatteryLevelEnoughForUpdate(batteryLevel)) {
```
[the remark this answers is not in the document]
Comment (wojcik92michal) [reply packages/app/src/update/actions/download-updates/download-updates.action.ts:39]: As far as I remember, this is the business requirement.

File: packages/app/src/__deprecated__/renderer/styles/theming/theme.ts (hunk 1 above)
Comment (wojcik92michal) [reply packages/app/src/__deprecated__/renderer/styles/theming/theme.ts:47]: This is the existing color from our design system (https://www.figma.com/file/82xELWhdkYtGeDQRkxYnf3/Center---Design-System?node-id=7%3A84&t=8xAbvJsOClb71HO8-4 - see `Grey 1`), the same as in `actionHover`. I didn't want to use `actionHover` for that - the name sounds inappropriate to what I wanted to achieve (just give the color to the text according to the design).

The above link does not specify text colors, but rather font types.

If you disagree, I can contact with design team and ask them to change the color, but seems to me that it follow our rules. What do you think?

File: packages/app/src/overview/components/system/system.component.tsx (hunk 2 above)
Comment (wojcik92michal) [reply packages/app/src/overview/components/system/system.component.tsx]: Under the hood, `onClick` was adding `event` to the function (which accepted optional parameter - Typescript was ok with that, but it was causing some unexpected problems). But you are right, this change seems suspicious. Decided to:
* revert the change and apply it higher
* provide small refactor of `ButtonTogglerItem` - now it provides TS types

File: packages/app/src/update/actions/download-updates/download-updates.action.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      return rejectWithValue(
+        new AppError(
+          UpdateError.TooLowBattery,
+          "Device has too low battery level"
+        )
+      )
+    }
+
+    const isActionCanelledByUser = (
```
[the remark this answers is not in the document]
Comment (wojcik92michal) [reply packages/app/src/update/actions/download-updates/download-updates.action.ts]: Yep, it definitely makes sense. Changed in https://github.com/mudita/mudita-center/pull/1064/commits/72a8d4d155b77f8b151edd9f2e1621c8a47805ad

File: packages/app/src/update/actions/download-updates/download-updates.action.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkarski) [reply packages/app/src/update/actions/download-updates/download-updates.action.ts:41]: ok, so please skip right now my comment 👍

File: packages/app/src/__deprecated__/renderer/styles/theming/theme.ts (hunk 1 above)
Comment (dkarski) [reply packages/app/src/__deprecated__/renderer/styles/theming/theme.ts:47]: So maybe renaming `tabHover` will be a good compromise? 🤔

**Review by dkarski — APPROVED** [review]
